examples/cooperative_preempt.rs

```diff
@@ -18,40 +18,40 @@ fn main() {
     //
     // There are two ways of yielding control:
     //
-    //  * Local::yield_if_needed(), which will yield if the task has run for too
-    //    long. What "too long" means is an implementation detail, but it will be
-    //    always somehow related to the latency guarantees that the task queues want
-    //    to uphold in their `Latency::Matters` parameter (or
+    //  * glommio::executor().yield_if_needed(), which will yield if the task has
+    //    run for too long. What "too long" means is an implementation detail, but
+    //    it will be always somehow related to the latency guarantees that the task
+    //    queues want to uphold in their `Latency::Matters` parameter (or
     //    Latency::NotImportant).
     //
-    //  * Local::later(), which will yield immediately (execute the rest of the
-    //    function later).
+    //  * glommio::executor().later(), which will yield immediately (execute the
```

Comment:
BTW, the wording on later's doc has to be updated. It moves the entire task queue out of the executor (to move just the current task you can call directly `futures_lite::yield_now()`, which is standard Rust.

That also probably means it could use a name change. `later()` is what seastar does, but it doesn't mean much. (And seastar's later is closer to futures_lite::yield_now(), it doesn't yield the entire task queue). 

Suggestions ?

Reply:
Tokio uses `coop()` as in "cooperate" which I believe works here. wdyt?

Reply:
I don't like `coop`. That is a lot similar in principle to `yield_if_needed`. It will count however many times you executed and make a decision based on that. This doesn't: it is unconditional. You call `later()`, it will yield.

What we should communicate instead is that unlike `yield_now`, the entire task queue is being yielded. Something like `yield_now_but_the_entire_task_queue_not_just_the_task`. Except of course, with a nicer name

Reply:
What about `yield_task_queue()`?

Reply:
I am between that an `yield_task_queue_now`. Let's give some time for other people to comment

Reply:
I think I like yield_task_queue might be descriptive enough.. what additional context are we getting from the "now" ? Maybe I'm just missing it..

Reply:
This is what I ended up doing:
* I added a `yield_now` function that calls `futures_lite::future::yield_now()` internally. I wanted to call it `yield` but it's a reserved keyword.
* I renamed `later` to `yield_task_queue_now`. I kept the `_now` to be consistent with `yield_now`.
* I updated the documentation to make it very clear what yielding a task queue vs a task mean.